src/Learning/MHHistograms/MHDiscretizerMODLHistogram.cpp

```diff
@@ -109,33 +111,50 @@ void MHDiscretizerMODLHistogram::DiscretizeValues(ContinuousVector* cvSourceValu
 			oaResultHistograms.Add(postprocessedOptimizedHistogram);
 		oaResultHistograms.Add(optimizedHistogram);
 
-		// Recherche du meilleur histogramme
-		if (postprocessedOptimizedHistogram != NULL)
-			bestHistogram = postprocessedOptimizedHistogram;
-		else
-			bestHistogram = optimizedHistogram;
-
-		// Cas avec contrainte sur le nombre max d'intervalles
+		// Les histogrammes MODL sont optimaux.
+		// Pour chaque variable numerique, on produit une serie d'histogrammes interpretables, par granularite croissante.
+		// Le probleme est que l'histogramme interpretable le plus fin est parfois tres complexe, de type "herisson",
+		// ce qui peut derouter un utilisateur non expert. Ce probleme se produit dans environ 20% des cas.
+		// Plutot que de montrer par defaut l'histogramme interpretable le plus fin, on propose d'utiliser la
+		// regle de Terell-Scott  (cf. https://en.wikipedia.org/wiki/Histogram), avec un nombre de bins de (2 N)^1/3
+		// Ce choix permet d'avoir par defaut un histogramme a la fois fin et plus simple a interpreter:
+		// - en evitant des choix trop heuristiques de type "regle du coude", ou nombre maximal de "peaks"
+		// - pas trop parcimonieux (comme la regle de Sturges en log(N))
+		// - ne dependant pas des valeurs (comme la regle de Scott, qui utilise l'ecart type)
+		// - tres simple a calculer
+		// - avec une justification theorique, facile a defendre
+		nTerellScottMaxPartNumber = int(ceil(pow(2 * cvSourceValues->GetSize(), 1.0 / 3)));
+		nMaxPartNumber = nTerellScottMaxPartNumber;
+
+		// La regle de Terell-Scott est pertinente pour eviter les histogrammes de type "herisson" comportant des
+		// centaines, voire des milliers d'intervalles: cela ameliore vraiment le choix de la granularite par defaut.
+		// Par contre, dans le cas des discretisation avec faible nombre d'intervalles, l'application de cette
+		// regle aboutit parfois a un choix de granulatite trop grossiere, avec perte d'information
```

Comment:
"// ... a un choix de granula**r**ité" (replace "t" with "r" where highlighted in boldface).